arbor/include/arbor/simd/vsx.hpp

```diff
@@ -0,0 +1,528 @@
+#pragma once
+
+// VSX SIMD intrinsics implementation.
+#if defined(__VEC__) || defined(__ALTIVEC__) || defined(__VSX__)
+#include <altivec.h>
+#include <cmath>
+#include <cstdint>
+
+#include <iostream>
+#include <arbor/simd/approx.hpp>
+#include <arbor/simd/implbase.hpp>
+
+namespace arb {
+    namespace simd {
+        namespace detail {
+
+            struct vsx_double4;
+            struct vsx_int4;
+            struct vsx_bool2x2;
+
+            template <>
+            struct simd_traits<vsx_double4> {
+                static constexpr unsigned width = 4;
```

Comment:
The problem was that arbor uses i32 and f64 types, so to make vector widths homogeneous I had to decide between 'wasting' two lanes of 4xi32 and using a native 4xi32 plus an array of two 2xf64.

Reply:
I understand. I would still prefer that we follow the design of the other backends and have the width be tied to the number of `double`s that can fit in a register as opposed to the number of `int`s. Otherwise it is misleading that a neon vector has a simd width half the size of a `vsx` vector despite them both having a physical size of 128 bits. Do you know whether making the change would affect the performance of the backend?

Reply:
I did some performance tests in the early phases, which indicated the 4xf64 to be (slightly) better. I assume it helps the compiler to unroll a bit more. However, I do not have numbers on the current implementation.

Reply:
If it's only slightly beneficial, I vote we use a width of 2.